src/serde_macros.rs

```diff
@@ -64,11 +64,9 @@ macro_rules! serde_impl(
                             E: ::serde::de::Error,
                         {
                             if v.len() != $t::LEN {
-                                Err(E::invalid_length(v.len(), &stringify!($len)))
+                                Err(E::invalid_length(v.len(), &format!("{}", $t::LEN).as_str()))
```

Comment:
AFAIK `stringify!` produces a string literal while this code will allocate a string

Reply:
Correct. Not sure how rustc can optimize that allocation away as it knows the `LEN` is constant. Anyway I thought since it's in an error path, performance was not so important. I could change this back with the effect that users of `hash_newtype` need to mention the hash length in the macro. Which I'm also fine with, btw. Let me know if you prefer this reverted.

Reply:
I'd prefer it reverted

Reply:
Actually, I'm not so concerned about allocations in error paths ... it's the other change that I want reverted, but I see that that's difficult.

Reply:
Reverted.